psydac/ddm/cart.py

```diff
@@ -91,8 +91,11 @@ def __init__( self, npts, pads, periods, reorder, comm=MPI.COMM_WORLD ):
         # ...
         # Know the number of processes along each direction
 #        self._dims = MPI.Compute_dims( self._size, self._ndims )
-        mpi_dims, block_shape = mpi_compute_dims( self._size, npts, pads )
-        self._dims = mpi_dims
+        if nprocs is None:
+            nprocs, block_shape = mpi_compute_dims( self._size, npts, pads )
```

Comment:
If the`nprocs` argument is given, can you check that it it contains something that makes sense?
For example, we should check that `len(nprocs) == len(npts)` (see assert at line 68)

Reply:
yes we should add it